### PR TITLE
SensorAPI

### PR DESCRIPTION
**Pull request by duongnln96**
Sensor API need to be reviewed

### CODE REVIEW

File: sensors/sensorlib/sensor_app.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        sensorApp.updateConfigSensor(sensorConfig)
+
+        return marshal(sensorConfig, sensorConfigFields)
+            
+
+class SensorApplication:
+    def __init__(self, sensorConfig: sensorConfigType):
+        self.__connector: SimpleRMQTopicConnection = SimpleRMQTopicConnection(RMQConfig(
+            "rabbitmq3",
```
Comment (tinhtn1508) [inline sensors/sensorlib/sensor_app.py]: Don't hard code the parameters of RMQConfig. You should get them from json or yml  file

File: sensors/sensorlib/sensor_app.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        self.sensorConfig = config
+        self.sensor = Sensor(SensorApplication.signalTypeConfig(self.__sensorConfig['sensorType']), self.__sensorConfig['cycle']).\
+                                    noise(NoiseConfig(self.__sensorConfig['noiseMean'], self.__sensorConfig['noiseStd'])).\
+                                    wave(WaveConfig(self.__sensorConfig['waveMagnitude'], self.__sensorConfig['wavePoint'])).\
+                                    callTo(lambda x: self.__connector.send(x)).\
+                                    initialize()
+        self.startSensor()
+
+    @classmethod
+    def app(cls) -> Any:
```
Comment (tinhtn1508) [inline sensors/sensorlib/sensor_app.py]: returned value is not "Any". May be "SensorApplication"

File: sensors/sensorlib/sensor_app.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('sensorType', type=str, required=True, location='json')
+        self.reqparse.add_argument('cycle', type=float, required=True, location='json')
+        self.reqparse.add_argument('noiseMean', type=float, required=True, location='json')
+        self.reqparse.add_argument('noiseStd', type=float, required=True, location='json')
+        self.reqparse.add_argument('waveMagnitude', type=float, required=True, location='json')
+        self.reqparse.add_argument('wavePoint', type=int, required=True, location='json')
+        super(SensorConfigAPI, self).__init__()
+
+    def get(self):
```
Comment (tinhtn1508) [inline sensors/sensorlib/sensor_app.py]: I saw you have already supported the type hints, please update

File: sensors/sensorlib/sensor_app.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    @staticmethod
+    def signalTypeConfig(typeStr: str) -> SignalType:
+        if typeStr == 'SineWave':
+            return SignalType.SINEWAVE
+        elif typeStr == 'StaticWave':
+            return SignalType.STATICWAVE
+        elif typeStr == 'SquareWave':
+            return SignalType.SQUAREWAVE
+
+sensorApp = SensorApplication.app()
```
Comment (tinhtn1508) [inline sensors/sensorlib/sensor_app.py]: Don't share "sensorApp" object to everywhere. 
1. Make this class to singleton
2. Forward it by a specific method

File: sensors/sensorlib/sensor_app.py (hunk 1 above)
Comment (tuantran1810) [reply sensors/sensorlib/sensor_app.py]: +1 move nó ra config nha fence

File: sensors/sensorlib/sensor_app.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            "measurment",
+            "measurement.sensors.sensor1",
+            MessageType.JSON,
+        )) 
+
+        self.__sensorConfig:  sensorConfigType = sensorConfig
+        self.__sensor: Sensor = Sensor(SensorApplication.signalTypeConfig(self.__sensorConfig['sensorType']), self.__sensorConfig['cycle']).\
+                                    noise(NoiseConfig(self.__sensorConfig['noiseMean'], self.__sensorConfig['noiseStd'])).\
+                                    wave(WaveConfig(self.__sensorConfig['waveMagnitude'], self.__sensorConfig['wavePoint'])).\
+                                    callTo(lambda x: print(x)).\
```
Comment (tuantran1810) [inline sensors/sensorlib/sensor_app.py]: cái này chỉ đang print chứ chưa lên rabbitmq nè fence

File: sensors/sensorlib/sensor_app.py [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+DEFAULT_SENSOR_CONFIG: sensorConfigType = {
+    'sensorType': 'StaticWave',
+    'cycle': 10.0,
+    'noiseMean': 0.0,
+    'noiseStd': 0.3,
+    'waveMagnitude': 10.0,
+    'wavePoint': 20    
+}
+
+sensorConfig: sensorConfigType = DEFAULT_SENSOR_CONFIG
```
Comment (tuantran1810) [inline sensors/sensorlib/sensor_app.py:51]: gói mấy cái này vô class hết nha fence

File: sensors/sensorlib/sensor_app.py (hunk 4 above)
Comment (tuantran1810) [reply sensors/sensorlib/sensor_app.py]: +1 đồng quan điểm

**Review by tuantran1810 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**